control/controldisplay/formatter_template.go

```diff
@@ -0,0 +1,169 @@
+package controldisplay
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/turbot/steampipe/control/controlexecute"
+	"github.com/turbot/steampipe/filepaths"
+)
+
+var ErrAmbiguousTemplate = errors.New("ambiguous templates found")
+var ErrTemplateNotFound = errors.New("template not found")
+
+type TemplateFormatter struct {
+	outputExtension string
+	template        *template.Template
+	name            string
+}
+
+func (tf TemplateFormatter) Format(ctx context.Context, tree *controlexecute.ExecutionTree) (io.Reader, error) {
+	reader, writer := io.Pipe()
+	go func() {
+		if err := tf.template.ExecuteTemplate(writer, "outlet", tree); err != nil {
+			writer.CloseWithError(err)
+		} else {
+			writer.Close()
+		}
+	}()
+	return reader, nil
+}
+
+func (tf TemplateFormatter) FileExtension() string {
+	extension := strings.TrimPrefix(tf.outputExtension, ".")
+
+	if extension == tf.name {
+		return extension
+	} else {
+		return fmt.Sprintf("%s.%s", tf.name, extension)
+	}
+}
+
+func CreateTemplateFormatter(input ExportTemplate) (*TemplateFormatter, error) {
+	t, err := template.New("outlet").
+		Funcs(formatterTemplateFuncMap).
+		ParseFS(os.DirFS(input.TemplatePath), "*")
+
+	if err != nil {
+		return nil, err
+	}
+	return &TemplateFormatter{name: input.FormatName, outputExtension: input.OutputExtension, template: t}, nil
+}
+
+func GetExportTemplate(export string, allowFilenameEvaluation bool) (format *ExportTemplate, filename string, err error) {
```

Comment:
// comment to describe the args

Reply:
describe purpose of ``allowFilenameEvaluation`

Reply:
done